src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs

```diff
@@ -89,26 +88,33 @@ public IPrivateConsoleDispatcher Dispatcher
             }
         }
 
+        [Obsolete]
         public IVsUIShell VsUIShell
-        {
-            get { return ServiceProvider.GetService<IVsUIShell>(typeof(SVsUIShell)); }
-        }
-
-        private IVsStatusbar VsStatusBar
         {
             get
             {
-                if (_vsStatusBar == null)
+                IVsUIShell svc = null;
+                ThreadHelper.JoinableTaskFactory.Run(async delegate
                 {
-                    _vsStatusBar = ServiceProvider.GetService<IVsStatusbar>(typeof(SVsStatusbar));
-                }
-                return _vsStatusBar;
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    svc = await AsyncServiceProvider.GlobalProvider.GetServiceAsync<IVsUIShell>();
+                });
+                return svc;
             }
         }
 
         private IOleServiceProvider OleServiceProvider
         {
-            get { return ServiceProvider.GetService<IOleServiceProvider>(typeof(IOleServiceProvider)); }
+            get
+            {
+                IOleServiceProvider svc = null;
+                ThreadHelper.JoinableTaskFactory.Run(async delegate
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    svc = await AsyncServiceProvider.GlobalProvider.GetServiceAsync<IOleServiceProvider>();
```

Comment:
This property is private. Change it to an async method instead and update all references. Using `JTF.Run` to run async code in a sync method just uses two threads instead of one, therefore it doesn't solve any problems. It's only useful when you have a synchronous public API that others are using that you need to keep supporting. Its an anti-pattern in private/internal code.

Reply:
If I understand correctly since code which is calling this one is itself is synchronous we better to keep old GetService instead of GetServiceAsync, is it correct? Otherwise we're just wrapping async code with JTF only increase overhead, but it still run as synchronous in the end.
https://github.com/NuGet/NuGet.Client/blob/2c0c7c3f220833d2438c552df3cabeab24872297/src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs#L141-L142

Reply:
yes, that's right. There's no benefit to doing `JTF.Run(async delegate { /* some anonymous method */})`.

`JTF.Run(...)` is suitable when you add a new async method, and for backwards compatibility reason you make the old sync method call the new async one, while API customers migrate to the new async method. But that's not happening here.

Reply:
Ok. Reverted this change.